### PR TITLE
Fix NRE in HighlightingService

### PR DESCRIPTION
**Pull request by genlu**
Internal bug [763988](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/763988)

@dotnet/roslyn-ide @CyrusNajmabadi

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlightingService.cs [hunk 1]
```diff
@@ -27,6 +27,11 @@ internal class HighlightingService : IHighlightingService
         public IEnumerable<TextSpan> GetHighlights(
              SyntaxNode root, int position, CancellationToken cancellationToken)
         {
+            if (root == null)
```
Comment (genlu) [inline src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlightingService.cs]: Couldn't figure out why this was `null` from the dump though.

Comment (mavasani) [reply src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlightingService.cs]: Tagging @heejaechang - related to https://github.com/dotnet/roslyn/pull/32388? Are both of them same underlying issues (root == null?)

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlightingService.cs]: definitely seems unfortunate that we don't know why.  that said, the fix seems reasonable.

Comment (genlu) [reply src/EditorFeatures/Core/Implementation/KeywordHighlighting/HighlightingService.cs]: Ah, seems @heejaechang was trying to address the same bug :)

**Comment by genlu** [comment]
@jasonmalinowski I'm interested in how did you figure this out with the dump, get a few minutes to show me? :)

**Comment by jasonmalinowski** [comment]
@genlu Remind me tomorrow, but as a warning it took @heejaechang and I pairing at a computer for an hour using both Visual Studio _and_ WinDBG. :smile:

**Comment by genlu** [comment]
Closing in favor of #32388

**Pull request closed without merging**